code/modules/mob/mob_grab.dm

```diff
@@ -170,7 +170,7 @@
 		if("mouth")
 			if(announce)
 				user.visible_message("<span class='warning'>\The [user] covers [target]'s mouth!</span>")
-			if(target.silent < 3)
+			if(target.silent < 3 && !isipc(target))

```

Comment:
Bad fix.
This just prevents the silencing but does not prevent the message.

Aside from that I would just change the message to covers targets "speaker", since the mechanic should also be possible for IPCs.